weeks/week02/HW2_Task2.py

```diff
@@ -0,0 +1,16 @@
+cache = {}
+def square(n):
+     if n in cache:
+         print(f'{cache[n]} cached')
+         return cache[n]
+
+     result = (n ** 2)
+     cache[n] = result
+     print(f'{cache[n]} culculated')
+     return result
+
+
+my_list = [2, 2, 3, 3, 4]
+squared_list = [square(num) for num in my_list]
```

Comment:
Nice that you used the lambda function here but it is not part of the exercise. You are squaring the numbers twice (one from this lambda function and another one from the method.